client/state/data-layer/wpcom-http/index.js

```diff
@@ -37,19 +39,35 @@ const queueRequest = ( { dispatch }, action, next ) => {
 		onProgress,
 		path,
 		query = {},
+		dedupe,
 	} = action;
 
 	const method = rawMethod.toUpperCase();
 
-	const request = fetcherMap( method )( ...compact( [
+	/* Eat the request under the following conditions:
+	 * 1. Dedupe flag is set true
+	 * 2. It is a GET request
+	 * 3. There is a currently ongoing request for the same path with the same query
+	 */
+	if ( method === 'GET' && dedupe && inflight.requestInflight( requestKey( action ) ) ) {
+		return;
+	}
+
+	const baseRequest = fetcherMap( method )( ...compact( [
 		{ path, formData },
 		query,
 		method === 'POST' && body,
-		( error, data ) => !! error
-			? onFailure && dispatch( extendAction( onFailure, failureMeta( error ) ) )
-			: onSuccess && dispatch( extendAction( onSuccess, successMeta( data ) ) )
 	] ) );
 
+	const request = method === 'GET' && dedupe
+		? inflight.promiseTracker( requestKey( action ), baseRequest )
+		: baseRequest;
+
+	request.then(
```

Comment:
the reason here is that if we don't supply a callback then `wpcom.js` will return a `Promise` instead of the actual `request` object and then we won't be able to do things like attach an `onprogress` handler or cancel the request